fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/service/MdcAdapter.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.fineract.infrastructure.core.service;
+
+import java.util.Map;
+import org.slf4j.MDC;
+import org.slf4j.spi.MDCAdapter;
+
+public class MdcAdapter implements MDCAdapter {
```

Comment:
I understand the concept but this is most definitely not going to work. On one hand the adapter is not used by anything and on the other, you're calling the MDC inside the adapter which'll end up in a recursion even if it's configured for SLF4J. The approach I suggested before would work.